module/zfs/vdev.c

```diff
@@ -1346,7 +1346,15 @@ vdev_open(vdev_t *vd)
 		vd->vdev_asize = asize;
 		vd->vdev_max_asize = max_asize;
 		if (vd->vdev_ashift == 0)
-			vd->vdev_ashift = ashift;
+			vd->vdev_ashift = ashift; /* use detected value */
+		else {
```

Comment:
cstyle: Normally please add `{ }` to the `if` portion of this kind of conditional.  That said, in this case we should do the range change on `vd->vdev_ashift` regardless of how it was set.